### PR TITLE
[improvement](Block) Replace Block(const PBlock&) with deserialize because it has heavy operations in ctor

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/tablets_channel.cpp [hunk 1]
```diff
@@ -436,9 +436,10 @@ Status TabletsChannel::add_batch(const PTabletWriterAddBlockRequest& request,
         }
     }
 
-    auto get_send_data = [&]() { return vectorized::Block(request.block()); };
+    vectorized::Block send_data;
+    auto get_send_data = [&]() { return send_data.deserialize(request.block()); };
```
Comment (yiguolei) [inline be/src/runtime/tablets_channel.cpp:440]: should create a new block and then return it.
if the get_send_data is called multiple times then it is wrong.

Comment (yiguolei) [inline be/src/runtime/tablets_channel.cpp:440]: why do we need this lambda???

File: be/src/vec/aggregate_functions/aggregate_function_sort.h [hunk 2]
```diff
@@ -87,7 +87,7 @@ struct AggregateFunctionSortData {
 
         PBlock pblock;
         pblock.ParseFromString(data);
-        block = Block(pblock);
```
Comment (yiguolei) [inline be/src/vec/aggregate_functions/aggregate_function_sort.h:90]: It is not equal.
Maybe should create new block and deserialize and call operator =

File: be/src/runtime/tablets_channel.cpp (hunk 1 above)
Comment (jacktengg) [reply be/src/runtime/tablets_channel.cpp:440]: historical reasons

**Comment by jacktengg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.09 seconds
 stream load tsv:          549 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162037145 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**